src/core/event_engine/state.rs

```diff
@@ -0,0 +1,48 @@
+use crate::{
+    core::event_engine::{EffectInvocation, Event, Transition},
+    lib::alloc::vec::Vec,
+};
+
+/// State machine state trait.
+///
+/// For transition, the state machine needs to know which effects should be
+/// dispatched during transition to target state in response to a specific
+/// event.
+///
+/// Types which are expected to be used as states should implement the trait.
+pub(crate) trait State {
```

Comment:
Do we need that trait?

Reply:
Yup. 
This is the base trait for other types of state machine `state` (right now only _subscribe_, but there will be more).